src/mjolnir.py

```diff
@@ -87,9 +87,12 @@ def _on_collect_status(  # noqa: C901
             logger.debug(
                 "Peer relation found, checking if is main unit before configuring Mjolnir"
             )
-            main_unit_id = peer_relation[0].data[self._charm.app].get("main_unit_id")
+            main_unit_id = (
+                peer_relation[0].data[self._charm.app].get("main_unit_id", self._charm.unit.name)
+            )
             if not self._charm.unit.name == main_unit_id:
                 logger.info("This is not the main unit, skipping Mjolnir configuration")
+                return
```

Comment:
Do you think an unit test might be worth for this?

Reply:
Done